File: 4 - Vetores/vetores3.php
<?php

# Escreva um programa que leia uma matriz N x M, e imprima a sua transposta.

$matriz = array(array(1, 2, 3), array(4, 5, 6));
$transposta = array();

echo "MATRIZ: <br>";
for ($i=0; $i < count($matriz); $i++) {
  for ($j=0; $j < count($matriz[$i]); $j++) {
    echo $matriz[$i][$j] . " ";
  }
  echo "<br>";
}

for ($i=0; $i < count($matriz); $i++) {
  for ($j=0; $j < count($matriz[$i]); $j++) {
    $transposta[$j][$i] = $matriz[$i][$j];
  }
}

echo "<br>";
echo "TRANSPOSTA: <br>";
for ($i=0; $i < count($transposta); $i++) {
  for ($j=0; $j < count($transposta[$i]); $j++) {
    echo $transposta[$i][$j] . " ";
  }
  echo "<br>";
}

File: 4 - Vetores/Ruby/vetores3.rb
# Escreva um programa que leia uma matriz N x M, e imprima a sua transposta.

matriz = []
transposta = []

puts "Digite a quantidade de linhas da matriz: "
linhas = gets.to_i

linhas.times do
    matriz << []
    transposta << []
end

for l in 0..matriz.size - 1 do
	for c in 0..matriz.size - 1 do
		matriz[l][c] = gets.to_i
	end
end

print "#{matriz}\n\n"

for l in 0..matriz.size - 1 do
	for c in 0..matriz.size - 1 do
		transposta[c][l] = matriz[l][c]
	end
end

print "#{transposta}"
